met-web/src/components/engagement/form/EngagementFormTabs/AdditionalDetails/AdditionalDetailsContext.tsx

```diff
@@ -0,0 +1,85 @@
+import React, { createContext, useContext, useState } from 'react';
+import { ActionContext } from '../../ActionContext';
+import { EngagementTabsContext } from '../EngagementTabsContext';
+import { useAppDispatch } from 'hooks';
+import { openNotification } from 'services/notificationService/notificationSlice';
+import { SubmissionStatus } from 'constants/engagementStatus';
+
+export interface AdditionalDetailsContextState {
+    initialConsentMessage: string;
+    setInitialConsentMessage: (richContent: string) => void;
+    consentMessage: string;
+    setConsentMessage: (richContent: string) => void;
+    handleSaveAdditional: () => void;
+    updatingAdditional: boolean;
+    hasBeenOpened: boolean;
+}
+
+export const AdditionalDetailsContext = createContext<AdditionalDetailsContextState>({
+    initialConsentMessage: '',
+    setInitialConsentMessage: () => {
+        return;
+    },
+    consentMessage: '',
+    setConsentMessage: () => {
+        return;
+    },
+    handleSaveAdditional: () => {
+        return;
+    },
+    updatingAdditional: false,
+    hasBeenOpened: false,
+});
+
+export const AdditionalDetailsContextProvider = ({ children }: { children: React.ReactNode }) => {
+    const { handleUpdateEngagementRequest, savedEngagement } = useContext(ActionContext);
+    const { engagementFormData } = useContext(EngagementTabsContext);
+    const dispatch = useAppDispatch();
+
+    const [updatingAdditional, setUpdatingAdditional] = useState(false);
+    const [consentMessage, setConsentMessage] = useState(savedEngagement?.consent_message || '');
+    const [initialConsentMessage, setInitialConsentMessage] = useState(savedEngagement?.consent_message || '');
+
+    const handleUpdateEngagementAdditional = () => {
+        return handleUpdateEngagementRequest({
+            ...engagementFormData,
+            consent_message: consentMessage,
+        });
+    };
+
+    const handleSaveAdditional = async () => {
+        try {
+            if (!savedEngagement.id) {
+                dispatch(openNotification({ text: 'Must create engagement first', severity: 'error' }));
```

Comment:
Does this mean that a user will need to save the Engagement before saving the consent message?

Reply:
yes, similar like settings tab